sqlglot/dataframe/sql/dataframe.py

```diff
@@ -260,7 +260,9 @@ def _add_ctes_to_expression(cls, expression: exp.Select, ctes: t.List[exp.CTE])
     @classmethod
     def _get_outer_select_columns(cls, item: t.Union[exp.Expression, DataFrame]) -> t.List[Column]:
         expression = item.expression if isinstance(item, DataFrame) else item
-        return [Column(x) for x in expression.find(exp.Select).expressions]
+        select = expression.find(exp.Select)
+        assert select, "No select expression found"
```

Comment:
handle `select` being `None`

Reply:
you can just do

return [Column(x) for x in (expression.find(exp.Select) or exp.Select().expressions)]

Reply:
this will fail to type check because `expression.find(exp.Select)` is not iterable

Reply:
sorry, i over complicated it just do

for x in expression.find(exp.Select) or []